signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java

```diff
@@ -54,25 +56,74 @@ public MappedResults<ArtifactSigner> load(
 
-    final Pair<Map<Path, String>, Integer> configFilesMapPair =
+    final ConfigFileContent configFileContent =
         getNewOrModifiedConfigFilesContents(configsDirectory, fileExtension);
-    final Map<Path, String> configFilesContentMap = configFilesMapPair.getLeft();
-    final int configFileErrorCount = configFilesMapPair.getRight();
 
-    final String timeTaken =
-        DurationFormatUtils.formatDurationHMS(Duration.between(start, Instant.now()).toMillis());
     LOG.info(
         "Signer configuration metadata files read in memory {} in {}",
-        configFilesContentMap.size(),
-        timeTaken);
+        configFileContent.getContentMap().size(),
+        calculateTimeTaken(start));
+
+    final Instant conversionStartInstant = Instant.now();
+    // convert yaml file content to list of SigningMetadata
+    final MappedResults<SigningMetadata> signingMetadataResults =
+        convertConfigFileContent(configFileContent.getContentMap(), signerParser);
+
+    LOG.debug(
+        "Signing configuration metadata files converted to signing metadata {}",
+        signingMetadataResults.getValues().size());
+
+    final Collection<SigningMetadata> signingMetadata = signingMetadataResults.getValues();
+    // filter hashicorp signing metadata, we want them to be processed sequentially.
+    final Map<Boolean, List<SigningMetadata>> partitionedMetadata =
+        signingMetadata.stream()
+            .collect(
+                Collectors.partitioningBy(metadata -> !metadata.getType().equals("hashicorp")));
+    final MappedResults<ArtifactSigner> metadataResultParallel =
+        mapToArtifactSignerInParallel(partitionedMetadata.get(true), signerParser);
+    final MappedResults<ArtifactSigner> metadataResultSequential =
+        mapToArtifactSignerSequentially(partitionedMetadata.get(false), signerParser);
 
     final MappedResults<ArtifactSigner> metadataResult =
-        processMetadataFilesInParallel(configFilesContentMap, signerParser);
-    metadataResult.mergeErrorCount(configFileErrorCount);
+        MappedResults.merge(metadataResultParallel, metadataResultSequential);
+
+    // merge error counts of config file parsing errors ...
+    metadataResult.mergeErrorCount(signingMetadataResults.getErrorCount());
+    metadataResult.mergeErrorCount(configFileContent.getErrorCount());
+
+    LOG.info(
+        "Total Artifact Signer loaded via configuration files: {}\nError count {}\nTime Taken: {}.",
+        metadataResult.getValues().size(),
+        metadataResult.getErrorCount(),
+        calculateTimeTaken(conversionStartInstant));
+
     return metadataResult;
   }
 
-  private Pair<Map<Path, String>, Integer> getNewOrModifiedConfigFilesContents(
+  private MappedResults<SigningMetadata> convertConfigFileContent(
+      final Map<Path, String> contentMap, final SignerParser signerParser) {
+    final AtomicInteger errorCount = new AtomicInteger(0);
```

Comment:
This doesn't need to be an AtomicInteger it's modified sequentially in the stream below from the same thread

Reply:
It has to `AtomicInteger` as "variables used in lambda has to be final or effective final". Unless we convert it from stream to straight for loop.